### ISSUE TITLE
Jenkins package breaks nix-shell

### ISSUE THREAD

**Issue by chris-martin** [NixOS/nixpkgs#14137]
Nix-shell doesn't work if `jenkins` is in the package list.

```
> nix-shell -p jenkins
bash: source: /nix/store/mycmjb4nzyjwhf7q0n2iix26ywb6xszd-jenkins-1.653: cannot execute binary file
```

Version info:

```
> uname -a
Linux renzo 4.4.4 #1-NixOS SMP Thu Mar 3 23:10:04 UTC 2016 x86_64 GNU/Linux
> nixos-version
16.09pre78286.0bf8a1a (Emu)
> nix-env --version
nix-env (Nix) 1.11.2
> nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion
"16.09pre78848.9549257"
```

**Comment by gilligan** [comment]
I wonder why this has been closed. The problem still remains:
```
$ nix-shell -p jenkins
bash: PK: command not found
```
That is on OSX but I assume the problem is going to be the same on NixOS/Linux. Could someone please verify and then re-open this?

**Comment by bjornfor** [comment]
I verified that the issue exist on latest NixOS.

I think the problem is that the jenkins package installs the war as $out. Not as a file inside the directory $out, but simply as the _file_ $out.

**Comment by gilligan** [comment]
Would be easy enough to fix - What would be an idiomatic solution to this ?
```
mkdir -p $out/someDirectory
cp jenkins.war $out/someDirectory
```
What should `someDirectory` be ?

**Comment by joachifm** [comment]
@gilligan if I understood @bjornfor's comment, you'd simply do `mkdir -p $out && cp jenkins.war $out/`.

**Comment by gilligan** [comment]
@joachifm D'OH - okay.

**Comment by bjornfor** [comment]
I don't know if there is a canonical location for .war files. But whatever directory is chosen, beware that there are some places in nixpkgs that expects the "jenkins" attribute to be the war file itself. Those references must be fixed up in the same commit that moves the file.

**Comment by bjornfor** [comment]
A .war file is apparently a .jar file. And .jar files appear mostly in ```$out/lib/``` or ```$out/lib/<pkg>/``` in nixpkgs.

**Comment by bjornfor** [comment]
I have a commit that should fix this. I will push it when it's ready (tested).

**Comment by bjornfor** [comment]
Done: 4075c10a59af455db2012beb313453662f09dad0

**Comment by danbst** [comment]
Copied from commit comment:

> hm, tomcat expects .war files to be placed in `$out/webapps/`
>
> https://github.com/NixOS/nixpkgs/blob/a04a727/nixos/modules/services/web-servers/tomcat.nix#L248
>
>just a consistency note

**Comment by bjornfor** [comment]
Copied from commit comment:

> Good point. Fixed in 8c2d888 ("jenkins: move $out/{lib => webapps}/jenkins.war").

:-)
